jkube-kit/generator/webapp/src/main/java/org/eclipse/jkube/generator/webapp/handler/TomcatAppSeverHandler.java

```diff
@@ -48,9 +48,41 @@ private boolean hasTomcatMavenPlugin() {
 
   @Override
   public String getFrom() {
+    if (projectHasLegacyJavaeeDependencies()) {
+      return imageLookup.getImageName("tomcat9.upstream.docker");
+    }
     return imageLookup.getImageName("tomcat.upstream.docker");
   }
 
+  private boolean projectHasLegacyJavaeeDependencies() {
```

Comment:
I'm not really following the logic behind this (especially considering the kind of dependencies you list). Plus, this is not aligned with the issue description (it's the complete opposite).

Maybe we should start by clarifying why Tomcat 10 might not be applicable for "legacy" apps and in which cases we should fallback to 9 (or vice versa, as the issue suggests). In the issue, the opposite situation was described, a dependency towards a "newer" artifact made the project incompatible with Tomcat 9.

AFAIK Tomcat 10 is breaking because it uses jakarta implementations of the javax packaged APIs (javax ---> jakarta). However, a dependency towards the javax.mail API jar, doesn't imply that your project won't have a dependency to the implementation jar too (e.g. `com.sun.mail:javax.mail`), thus making it viable for Tomcat 10.

Reply:
Sorry, I did not mentioned where this was coming from, maybe it should be specified in a comment.
https://github.com/apache/tomcat-jakartaee-migration#differences-between-java-ee-8-and-jakarta-ee-9

You are absolutely right, we would need to check the usage of the import packages in the generated war.

For `com.sun.mail:javax.mail` it would work by chance because it has `javax.activation:activation` as a transitive dependency

Reply:
> For `com.sun.mail:javax.mail` it would work by chance because it has `javax.activation:activation` as a transitive dependency

What I'm trying to tell is that you can deploy into Tomcat10 webapp server any javax API based application if you include the implementation libraries too.

So maybe, instead of making this something `smart` we might just want to make it configurable.

Reply:
OK what about we just log a warning with the same condition here and suggest the user to use `jkube.generator.webapp.server=tomcat9` if any issue ?

Reply:
The main problem I see is that your condition is checking on group IDs and not on package imports.
Maybe we can just print an INFO message stating that Tomcat 10 was selected, and suggest the configuration property with a message like "If your project is still using javax.* imports..."

Reply:
This is interesting: 

> Tomcat can convert an existing web application from Java EE 8 to Jakarta EE 9 at deployment time using the Apache Tomcat migration tool for Jakarta EE. To make use of the feature, the web application should be placed in the Host legacyAppBase folder (by default named webapps-javaee) and they will be converted to an equivalent Jakarta EE web application in the Host appBase folder (by default named webapps).

from https://tomcat.apache.org/migration-10.html#Migrating_from_9.0.x_to_10.0.x

Reply:
> The main problem I see is that your condition is checking on group IDs and not on package imports. Maybe we can just print an INFO message stating that Tomcat 10 was selected, and suggest the configuration property with a message like "If your project is still using javax.* imports..."

I know it is not perfect. But this would give a good hint in 99% of the case.  Whereas if we show the message all the time, it would be misleading.
As a user I would prefer, everything automated with the log messages explaining the choices that had been made by JKube and the alternative possible options. So basically keep this condition that would detect the 99% right configuration with a log message that the webapp had been placed to `webapps-javaee` and that it is possible to place it in `webapp` if it doesn't fit.

Reply:
To summarize:
- war file is copied to the compat/migrate directory `webapps-javaee` (regardless of its dependencies) ---> Test if this works
- INFO message is printed indicating this, plus suggesting the config flag in case the application fails to start

Reply:
So if I understand correctly, we provide 2 configurations, one to set the tomcat version and another to set the deployment directory.
If any of this configurations is the default, then we log the INFO message on why this was chosen and how to override the configuration.
Right?

Reply:
I think we can drop the tomcat version. setting the deployment directory is good enough. Do we still need to keep `tomcat9` beside the previous JKube versions ?